src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V63/Messages/ReceiptsMessageSerializer.cs

```diff
@@ -14,7 +14,7 @@ namespace Nethermind.Network.P2P.Subprotocols.Eth.V63.Messages
 {
     public class ReceiptsMessageSerializer : IZeroInnerMessageSerializer<ReceiptsMessage>
     {
-        private static readonly RlpLimit RlpLimit = RlpLimit.For<ReceiptsMessage>(NethermindSyncLimits.MaxReceiptFetch, nameof(ReceiptsMessage.TxReceipts));
+        private static readonly RlpLimit RlpLimit = RlpLimit.For<ReceiptsMessage>(NethermindSyncLimits.MaxReceiptFetch * 4, nameof(ReceiptsMessage.TxReceipts));
```

Comment:
Looks like problem was here, not sure why are we receiving messages with such long receipts. And we actually get them during snap sync.